pkg/controller/deadmanssnitch/deadmanssnitch_controller.go

```diff
@@ -130,47 +126,9 @@ func (r *ReconcileDeadMansSnitch) Reconcile(request reconcile.Request) (reconcil
 	reqLogger.Info("Reconciling DeadMansSnitch")
 
 	// Fetch the ClusterDeployment instance
-	instance := &hivev1alpha1.ClusterDeployment{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// Request object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
-			// Return and don't requeue
-			return reconcile.Result{}, nil
-		}
-		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
-	}
-
-	// Just return if this is not a managed cluster OR has noalerts label set
-	if val, ok := instance.Labels[ClusterDeploymentManagedLabel]; ok {
-		if val != "true" {
-			reqLogger.Info("Not a managed cluster", "Namespace", request.Namespace, "Name", request.Name)
-			return reconcile.Result{}, nil
-		}
-	} else {
-		// Managed tag is not present which implies it is not a managed cluster
-		reqLogger.Info("Not a managed cluster", "Namespace", request.Namespace, "Name", request.Name)
-		return reconcile.Result{}, nil
-	}
-
-	// Cleanup DMS then return if alerts are disabled on the cluster
-	if _, ok := instance.Labels[ClusterDeploymentNoalertsLabel]; ok {
-		reqLogger.Info("Managed cluster with Alerts disabled", "Namespace", request.Namespace, "Name", request.Name)
-		return reconcile.Result{}, deleteDMS(r, request, instance, reqLogger)
-	}
+	processCD, instance, err := utils.CheckClusterDeployment(request, r.client, reqLogger)
```

Comment:
err looks unhandled here

Reply:
I have pushed an update to check and requeue.  The push is not showing up in the PR which is weird.  Hopefully it lands soon: https://github.com/jewzaam/deadmanssnitch-operator/commit/8f4d7ff04c2259951de6998fa184c39b3d6b116b

Reply:
Squashed and force pushed.  Now requeues if there is any error.